pandas/core/indexes/base.py

```diff
@@ -4973,6 +4993,22 @@ def _maybe_promote(self, other: "Index"):
 
         return self, other
 
+    def _get_other_deep(self, other: "Index") -> "Index":
```

Comment:
this should just be a function, no? its also a funny name. I think we have other similar things (e.g. this look like what .to_dense() does but for dtypes and not values).

Reply:
updated to _unpack_nested_dtypes function